### PR TITLE
Updates course deletion to new requirements 

### PR DESCRIPTION
**Pull request by dr15**
- Removes batch delete key.
- Prevents deleting a course that is published in any environment.

### CODE REVIEW

File: src/components/Screen/CoursesScreen/ActionsCell.tsx [hunk 1]
```diff
@@ -23,6 +27,22 @@ interface IActionsCell extends ITableCell {
...
+  getPublishedEnvironments(course, environments).length > 0;
+
+const getCannotDeleteText = (course: UICourse, environments: WalkMeEnvironment[]) => {
+  const publishedEnvs = getPublishedEnvironments(course, environments);
+  const publishedEnvsString = publishedEnvs.join(', ');
+
+  return (
+    <p>
+      This item is published in {pluralizer('environment', publishedEnvs.length)}:{' '}
+      <span className={classes.strong}>{publishedEnvsString}</span>. Please archive in all
```
Comment (dvira) [inline src/components/Screen/CoursesScreen/ActionsCell.tsx:40]: We normally write `classes['strong']`

File: src/components/Screen/CoursesScreen/ActionsCell.tsx [hunk 2]
```diff
@@ -31,14 +51,21 @@ export default function ActionsCell({
...
+      environments,
       dateRange: { from, to },
     },
   ] = useAppContext();
   const [{ isDeletingCourses }, dispatch] = useCoursesContext();
   const [showDeleteCourse, setShowDeleteCourse] = useState(false);
-  const isPublished =
-    course.publishStatus === PublishStatus.Published ||
-    course.publishStatus === PublishStatus.Modified;
+  const [canNotBeDeleted, setCanNotBeDeleted] = useState(getCannotBeDeleted(course, environments));
```
Comment (dvira) [inline src/components/Screen/CoursesScreen/ActionsCell.tsx:60]: I think 'cannot' is one word.

File: src/components/Screen/CoursesScreen/CoursesList.tsx [hunk 3]
```diff
@@ -220,12 +220,12 @@ function CoursesList({
...
                       disabled={selectedRowsOutOfScope}
                     />
-                    <DeleteCoursesButton
+                    {/*<DeleteCoursesButton
                       selectedRows={selectedRows}
                       isDeletingCourses={isDeletingCourses}
                       dispatch={dispatch}
                       disabled={selectedRowsOutOfScope}
-                    />
+                    />*/}
```
Comment (dvira) [inline src/components/Screen/CoursesScreen/CoursesList.tsx:228]: Please also comment out unused vars on lines 29 and 64.

File: src/components/Screen/CoursesScreen/style.module.scss [hunk 4]
```diff
@@ -1,5 +1,9 @@
 @import 'src/styles/mixins';
 
+.strong {
+  font-weight: bold;
```
Comment (dvira) [inline src/components/Screen/CoursesScreen/style.module.scss:4]: If that's all you need why not use `<strong>` instead of span?

**Review by dvira — CHANGES_REQUESTED** [review]
Looks good, left a few comments

File: src/components/Screen/CoursesScreen/ActionsCell.tsx (hunk 2 above)
Comment (dr15) [reply src/components/Screen/CoursesScreen/ActionsCell.tsx:60]: https://www.grammarly.com/blog/cannot-or-can-not/#:~:text=Can't%20is%20a%20contraction%20of%20cannot%2C%20and%20it's%20best,.%20.%20.%20but%20also.%E2%80%9D
Both are possible but I'll change it

File: src/components/Screen/CoursesScreen/CoursesList.tsx (hunk 3 above)
Comment (dr15) [reply src/components/Screen/CoursesScreen/CoursesList.tsx:228]: This mean I will also have to comment this out further up in the component that is using this one. Seems unnecessary to me as this is just temporary.

File: src/components/Screen/CoursesScreen/style.module.scss (hunk 4 above)
Comment (dr15) [reply src/components/Screen/CoursesScreen/style.module.scss:4]: Right you are

File: src/components/Screen/CoursesScreen/ActionsCell.tsx (hunk 1 above)
Comment (dr15) [reply src/components/Screen/CoursesScreen/ActionsCell.tsx:40]: As discussed when the class has an '-' we use it like this otherwise with dot notation

**Review by dvira — APPROVED** [review]

File: src/components/Screen/CoursesScreen/CoursesList.tsx (hunk 3 above)
Comment (dvira) [reply src/components/Screen/CoursesScreen/CoursesList.tsx:228]: Line 29 does not affect other components and is mentioned in the terminal.

File: src/components/Screen/CoursesScreen/ActionsCell.tsx (hunk 2 above)
Comment (dvira) [reply src/components/Screen/CoursesScreen/ActionsCell.tsx]: I meant in this file.

**Pull request merged**